components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -4553,4 +4564,28 @@ private static String resolveTenantDomain(HttpServletRequest request) {
         }
         return tenantDomain;
     }
+
+    /**
+     * Resolve the tenant domain from the application resident organization id which will be set when the resource
+     * is accessing from the tenanted endpoint.
+     *
+     * @return Application resident tenant domain.
+     * @throws FrameworkException When an error occurred while resolving the tenant domain.
+     */
+    public static String getAppResidentTenantDomain() throws FrameworkException {
```

Comment:
Do we need to have this as a public method?

Reply:
We can move this to a private method.